app/jobs/regular/digest_rag_upload.rb

```diff
@@ -9,17 +9,24 @@ class DigestRagUpload < ::Jobs::Base
     # TODO(roman): Add a way to automatically recover from errors, resulting in unindexed uploads.
     def execute(args)
       return if (upload = Upload.find_by(id: args[:upload_id])).nil?
-      return if (ai_persona = AiPersona.find_by(id: args[:ai_persona_id])).nil?
+
+      target_type = args[:target_type]
+      target_id = args[:target_id]
+
+      return if !target_type || !target_id
+
+      target = target_type.constantize.find_by(id: target_id)
```

Comment:
Maybe `rescue nil` here to be safe? `constantize` could return `NameError`.

Reply:
the tricky thing here is that this would be incorrect usage, I don't want to eat the error of a typo... the "blank target_type" is just there for the transition period.